docs/events/2023/docker-workshop.md

```diff
@@ -0,0 +1,235 @@
+---
+title: Docker Workshop
+description: Docs from our 2023 Docker Workshop event.
+date: 2023-08-25
+comments: true
+tags:
+  - Workshop
+---
+
+# Docker Workshop 2023
+
+## What is Docker?
+
+Docker is an awesome technology that allows you to run your applications in containers, cross platform without installing them permanently on your system.
+
+In this workshop we went over how `Dockerfiles` and `docker-compose` files work as well as how to build your own images and run them.
+
+![Containers vs VM](../../assets/images/workshops/2023-docker/containers-vs-virtual-machines.jpg)
+
+Docker containers and virtual machines have a lot in common, they both allow you to create a template for your application and run it on any system. The main difference is that docker containers are much more lightweight and can be run on any system that has docker installed.
+
+Instead of packaging an entire operating system like a virtual machine, docker containers only package the dependencies that your application needs to run. This means that you can run multiple docker containers on the same system without them interfering with each other.
+
+## Prerequisites
+
+### Docker
+
+There are a number of ways to install Docker but the easiest way to install it is docker desktop. Docker desktop is a GUI application that help you manage your docker images and containers without having to use the command line.
+
+### Dive (Optional)
+
+Dive is a piece of software that allows you to see the result of different steps in your docker image. It is a great tool to help you debug your docker images and see what is taking up the most space.
+
+
+
+## `Dockerfile`s
+
+Docker files are the core of the software. They allow you to define the install steps of your software and the different things that it needs like ports and storage.
+
+This workshop uses this [Example Code](https://github.com/ProgSoc/DockerWorkshop2023) to demo how docker files and docker compose files work and includes two different projects that need different steps to install. One easy and one harder to show the different ways that you can install software.
+
+### Basic `Dockerfile` Structure
+
+```dockerfile
+# Use an official Node.js runtime as the base image
+FROM node:20-alpine
```

Comment:
I think it's important to mention more in depth what the "FROM" command does, at least mentioning the tag structure (what node and : and 20-alpine are, linking to the dockerhub page for node, etc).

Could either mention it here or somewhere above or below, but not too far from this code block.